Helper/ConfigHelper.php

```diff
@@ -1024,85 +1124,53 @@ public function getSortingIndices($originalIndexName, $storeId = null, $currentC
-                    if (!array_key_exists('label', $newAttr) && array_key_exists('sortLabel', $newAttr)) {
-                        $newAttr['label'] = $newAttr['sortLabel'];
+                    $customerGroupId = (int) $group->getData('customer_group_id');
+                    if (!$this->isGroupPricingExcludedFromWebsite($customerGroupId, $websiteId)) {
+                        $newAttr = $this->getCustomerGroupSortPriceOverride($originalIndexName, $customerGroupId, $currency, $attr);;
+                        $attributesToAdd[$newAttr['sort']][] = $this->decorateSortAttribute($newAttr);
                     }
-                    $newAttr['ranking'] = [
-                        $newAttr['sort'] . '(' . $groupSortAttribute . ')',
-                        'typo',
-                        'geo',
-                        'words',
-                        'filters',
-                        'proximity',
-                        'attribute',
-                        'exact',
-                        'custom',
-                    ];
-                    $attributesToAdd[$newAttr['sort']][] = $newAttr;
                 }
-            } elseif ($attr['attribute'] === 'price') {
+            // Regular pricing
+            } elseif ($attr['attribute'] === self::SORT_ATTRIBUTE_PRICE) {
                 $indexName = $originalIndexName . '_' . $attr['attribute'] . '_' . 'default' . '_' . $attr['sort'];
                 $sortAttribute = $attr['attribute'] . '.' . $currency . '.' . 'default';
+            // All other sort attributes
             } else {
                 $indexName = $originalIndexName . '_' . $attr['attribute'] . '_' . $attr['sort'];
                 $sortAttribute = $attr['attribute'];
             }
+
+            // Decorate all non group pricing attributes
             if ($indexName && $sortAttribute) {
                 $attrs[$key]['name'] = $indexName;
-                if (!array_key_exists('label', $attrs[$key]) && array_key_exists('sortLabel', $attrs[$key])) {
-                    $attrs[$key]['label'] = $attrs[$key]['sortLabel'];
-                }
-                $attrs[$key]['ranking'] = [
-                    $attr['sort'] . '(' . $sortAttribute . ')',
-                    'typo',
-                    'geo',
-                    'words',
-                    'filters',
-                    'proximity',
-                    'attribute',
-                    'exact',
-                    'custom',
-                ];
+                $attrs[$key]['ranking'] = $this->getSortAttributingRankingSetting($sortAttribute, $attr['sort']);
+                $attrs[$key] = $this->decorateSortAttribute($attrs[$key]);
             }
         }
         $attrsToReturn = [];
-        if (count($attributesToAdd) > 0) {
-            foreach ($attrs as $key => $attr) {
-                if ($attr['attribute'] == 'price' && isset($attributesToAdd[$attr['sort']])) {
+        if (count($attributesToAdd)) {
```

Comment:
I gave it a try, and my replicas are not created anymore. I tried to have a look at this method and it seems that `$attrsToReturn` is only filled when `$attributesToAdd` is not empty (which seems to be the case if the customer groups are not enabled, which is the case for me and `getSortingIndices` is returning an empty array no matter the sorting config in the admin) . Maybe I misunderstood something in the process ?

Reply:
Good catch @damcou ! That issue is from legacy code that probably wasn't obvious before because of the multiple ways replicas were getting written back to Algolia so the needed were _evenutally_ created. My goal is to run it through one funnel and I've been mainly testing with customer groups so this is a helpful find.

I pushed a fix for this issue but it doesn't handle all scenarios. There is more work to do with regards to toggling virtual vs standard on an existing index along with customer group pricing enablement. There are separate tickets for these items which I will follow up in a future PR.